packages/twenty-front/src/modules/object-record/record-index/hooks/useLoadRecordIndexBoardColumn.ts

```diff
@@ -68,17 +69,38 @@ export const useLoadRecordIndexBoardColumn = ({
     recordBoardId,
   });
 
-  const recordIndexKanbanFieldMetadataItem = objectMetadataItem.fields.find(
-    (field) => field.id === boardFieldMetadataId,
-  );
+  let recordIndexKanbanFieldMetadataFilter: { [x: string]: any } | undefined;
+
+  if (isDefined(boardFieldMetadataId)) {
+    const recordIndexKanbanFieldMetadataItem = objectMetadataItem.fields.find(
+      (field) => field.id === boardFieldMetadataId,
+    );
+
+    if (!isDefined(recordIndexKanbanFieldMetadataItem)) {
+      throw new Error('Record index kanban field metadata item not found');
+    }
+
+    if (
+      !isDefined(recordIndexKanbanFieldMetadataItem?.name) ||
+      isEmpty(recordIndexKanbanFieldMetadataItem.name)
+    ) {
+      throw new Error('Record index kanban field metadata item name not found');
+    }
+
+    recordIndexKanbanFieldMetadataFilter = {
+      [recordIndexKanbanFieldMetadataItem.name]: isDefined(
+        recordGroupDefinition?.value,
+      )
+        ? { in: [recordGroupDefinition.value] }
+        : { is: 'NULL' },
+    };
+  }
 
   const filter = {
     ...requestFilters,
-    [recordIndexKanbanFieldMetadataItem?.name ?? '']: isDefined(
-      recordGroupDefinition?.value,
-    )
-      ? { in: [recordGroupDefinition?.value] }
-      : { is: 'NULL' },
+    ...(isDefined(recordIndexKanbanFieldMetadataFilter) && {
```

Comment:
Avoid && as it's not the same as using a ternary and it can return weird stuff, it can be tolerated in a JSX statement but not when creating objects.